RoaringBitmap/src/main/java/org/roaringbitmap/longlong/LongUtils.java

```diff
@@ -0,0 +1,82 @@
+package org.roaringbitmap.longlong;
+
+public class LongUtils {
+
+  //get the high 48 bit parts of the input data
+  public static byte[] highPart(long num) {
+    byte[] high48 = new byte[]{
+        (byte) ((num >>> 56) & 0xff),
+        (byte) ((num >>> 48) & 0xff),
+        (byte) ((num >>> 40) & 0xff),
+        (byte) ((num >>> 32) & 0xff),
+        (byte) ((num >>> 24) & 0xff),
+        (byte) ((num >>> 16) & 0xff)
+    };
+    return high48;
+  }
+
+  //get the low 16 bit parts of the input data
+  public static char lowPart(long num) {
+    return (char) num;
+  }
+
+  public static long toLong(byte[] high, char low) {
```

Comment:
I suggest extending the shims. On JDK9+ you can do this more efficiently with [`VarHandle`s](https://docs.oracle.com/javase/9/docs/api/java/lang/invoke/MethodHandles.html#byteArrayViewVarHandle-java.lang.Class-java.nio.ByteOrder-) and for JDK8 users it's safe to use `Unsafe`

Reply:
agree

Reply:
I am not familiar with this new API. I don't know how to use this API to accelerate the performance by assembling two different types array by looking at the docs. So I keep it.